src/http/one/TeChunkedParser.cc

```diff
@@ -139,13 +139,18 @@ Http::One::TeChunkedParser::parseChunkExtension(Http1::Tokenizer &tok, bool skip
             debugs(94, 5, "skipping unknown chunk extension " << ext);
 
             // unknown might have a value token or quoted-string
-            if (tok.quotedStringOrToken(value) && !tok.atEnd()) {
-                buf_ = tok.remaining(); // parse checkpoint
-                continue;
+            bool moreNeeded = false;
+            if (tok.quotedStringOrToken(value, moreNeeded)) {
+                if (!tok.atEnd()) {
```

Comment:
If this `if` statement and the following `break` statement can be removed without affecting the outcome, please remove them. An additional but rare attempt to parse (at the start of the loop) is better than this code duplication/complication. And it can be easily eliminated by other means.